csharp/src/lesson01/README.md

```diff
@@ -0,0 +1,301 @@
+# Lesson 1 - Hello World
+
+## Objectives
+
+Learn how to:
+
+* Instantiate a Tracer
+* Create a simple trace
+* Annotate the trace
+
+## Walkthrough
+
+### A simple Hello-World program
+
+Let's create a simple C# program `lesson01/exercise/Hello.cs` that takes an argument and prints `"Hello, {arg}!"`.
+
+```csharp
+using System;
+
+namespace OpenTracing.Tutorial.Lesson01.Exercise
+{
+    internal class Hello
+    {
+        public void SayHello(string helloTo)
+        {
+            var helloString = $"Hello, {helloTo}!";
+            Console.WriteLine(helloString);
+        }
+        public static void Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                throw new ArgumentException("Expecting one argument");
+            }
+
+            var helloTo = args[0];
+            new Hello().SayHello(helloTo);
+        }
+    }
+}
+```
+
+Run it:
+```powershell
+$ dotnet run Bryan
+Hello, Bryan!
+```
+
+### Create a trace
+
+A trace is a directed acyclic graph of spans. A span is a logical representation of some work done in your application.
+Each span has these minimum attributes: an operation name, a start time, and a finish time.
+
+Let's create a trace that consists of just a single span. To do that we need an instance of the `OpenTracing.ITracer`.
+We can use a global instance returned by `OpenTracing.Util.GlobalTracer.Instance`.
+
+```csharp
+using System;
+using OpenTracing.Util;
+
+namespace OpenTracing.Tutorial.Lesson01.Exercise
+{
+    internal class Hello
+    {
+        private readonly ITracer _tracer;
+
+        public Hello(OpenTracing.ITracer tracer)
+        {
+            _tracer = tracer;
+        }
+
+        public void SayHello(string helloTo)
+        {
+            var span = _tracer.BuildSpan("say-hello").Start();
+            var helloString = $"Hello, {helloTo}!";
+            Console.WriteLine(helloString);
+            span.Finish();
+        }
+
+        public static void Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                throw new ArgumentException("Expecting one argument");
+            }
+
+            var helloTo = args[0];
+            new Hello(GlobalTracer.Instance).SayHello(helloTo);
+        }
+    }
+}
+```
+
+We are using the following basic features of the OpenTracing API:
+  * a `ITracer` instance is used to create a span builder via `BuildSpan()`
+  * each `ISpan` is given an _operation name_, `"say-hello"` in this case
+  * builder is used to create a span via `Start()`
+  * each `ISpan` must be finished by calling its `Finish()` function
+  * the start and end timestamps of the span will be captured automatically by the tracer implementation
+
+However, if we run this program, we will see no difference, and no traces in the tracing UI.
+That's because the `GlobalTracer.Instance` returns a no-op tracer by default.
+
+### Use `Microsoft.Extensions.Logging` for output
```

Comment:
I would recommend moving this section before `### Create a trace`, because it breaks the flow of the actual tracing tutorial. Is it even necessary as a section? I think the very first code snippet can use logger right away, without much explanation since it's a common thing. If desired, you can mention it similar to L109-110.

Reply:
Using this logging framework is mostly common for ASP.NET Core but not in console apps. And not everyone knows it. So it’s just a bit of introduction since our jaeger client uses it internally